vpnglean/ios.cmake

```diff
@@ -0,0 +1,118 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# When building for iOS we want to include the Glean iOS SDK on the build as well.
+# Instead of using it out of the box, we want it to be linked to the vpnglean_ffi target,
+# so we generate the SDK oursevels.
+
+enable_language(Swift)
+
+set(GLEAN_VENDORED_PATH ${CMAKE_SOURCE_DIR}/3rdparty/glean)
+
+add_library(iosglean SHARED)
+
+target_include_directories(iosglean PUBLIC ${CMAKE_SOURCE_DIR})
+target_include_directories(iosglean PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(iosglean PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/glean)
+
+set_target_properties(iosglean PROPERTIES
+    FRAMEWORK TRUE
+    FRAMEWORK_VERSION C
+    OUTPUT_NAME "Glean"
+    MACOSX_FRAMEWORK_INFO_PLIST "${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Info.plist"
+    MACOSX_FRAMEWORK_IDENTIFIER "${BUILD_IOS_APP_IDENTIFIER}.Glean"
+    XCODE_ATTRIBUTE_SWIFT_VERSION "5.0"
+    XCODE_ATTRIBUTE_CLANG_ENABLE_MODULES "YES"
+    XCODE_ATTRIBUTE_APPLICATION_EXTENSION_API_ONLY "YES"
+    XCODE_ATTRIBUTE_MACH_O_TYPE "staticlib"
```

Comment:
For this one, why did it get included? I'm not sure we need to customize this here, though there may be a good reason to.

All the rest of these settings seem to match what we've done elsewhere in the project, but this one stands out.

That said, I do not believe this setting will hurt anything.

Reply:
Oh, this setting is actually super important. Without it we run into trouble when linking the iosglean framework with the main MozillaVPN project. To be honest I don't really understand why, this is the fruit of a lot of trial and error when implementing the prototype for this work :sweat_smile: